### ISSUE TITLE
Simple optimization problem running faster in Tensor Flow compared to JAX with Optax

### ISSUE THREAD

**Issue by alexandermm** [google/jax#16916]
### Description

Hi,

I've been trying to port Tensor Flow code in Google Colab (free version) to JAX but the execution time is 5 times slower:

The original TF code takes 4 seconds to run:
https://colab.research.google.com/drive/1HYYBr4M3hstzmPnTmYO8DEbodTuFBJb4?usp=drive_link

The equivalent JAX code takes 20 seconds to run:
https://colab.research.google.com/drive/1E9L_iyf7ZG0WfUgxmtbwAhTSuWdWzMei?usp=drive_link

I changed the optimizer to the JAX library Adam optimizer (jax.example_libraries) and it runs in 5 seconds but gives the wrong answer:
https://colab.research.google.com/drive/1dM0QeRWeOoMhejnyymtCqh_-EjVmCi8z?usp=sharing
(not sure why, the loss seems to decrease at the same rate as the Optax implementation)

I got the equivalent JAX code to run 5 times faster by taking the iteration step calculations, putting them in a function and jitting it but I haven't been able to accelerate the code. The loss and field calculation functions on their own are times faster in JAX than in TF.

Regards,
Alex

### What jax/jaxlib version are you using?

Latest in Colab

### Which accelerator(s) are you using?

GPU (in free Colab)

### Additional system info

Google Colab
